build_projects/shared-build-targets-utils/Utils/HostVersion.cs

```diff
@@ -19,24 +19,29 @@ public struct VerInfo
             public string Release;
             public string BuildMajor;
             public string BuildMinor;
+            public string CommitCountString;
 
-            public VerInfo(int major, int minor, int patch, string release, string buildMajor, string buildMinor)
+            public VerInfo(int major, int minor, int patch, string release, string buildMajor, string buildMinor, string commitCountString)
             {
                 Major = major;
                 Minor = minor;
                 Patch = patch;
                 Release = release;
                 BuildMajor = buildMajor;
                 BuildMinor = buildMinor;
+                CommitCountString = commitCountString;
             }
 
             public string GenerateMsiVersion()
             {
-                return Version.GenerateMsiVersion(Major, Minor, Patch, Int32.Parse(BuildMajor));
+                return Version.GenerateMsiVersion(Major, Minor, Patch, Int32.Parse(VerRsrcBuildMajor));
             }
 
             public string WithoutSuffix => $"{Major}.{Minor}.{Patch}";
 
+            public string VerRsrcBuildMajor => !string.IsNullOrEmpty(BuildMajor) ? BuildMajor : CommitCountString;
```

Comment:
NuGet versioning doesn't take a build major because of its notion of stable. But MSI version will be 1.0.1.3456. The reason is two different MSI binaries should not be 1.0.1.